Makefile

```diff
@@ -73,21 +86,85 @@ gateway: ## Generates grpc-gateway resources
 		--grpc-gateway_opt grpc_api_configuration=stroeer/page/article/v1/api_config_http.yaml \
 		stroeer/page/article/v1/article_page_service.proto
 
+.PHONY: clean
 clean: ## Deletes all generated files
 	@echo "+ $@"
 	rm -rf $(JAVA_DIR) || true
 	rm -rf `find $(GO_DIR) -type d \( -iname "*" ! -iname "go.mod" ! -iname "go.sum" ! -iname "*_test.go" \) -mindepth 1 -maxdepth 1` 2> /dev/null || true
 	rm -rf `find $(NODE_DIR) -type d \( -iname "*" ! -iname "node_modules" ! -iname "__tests__" \) -mindepth 1 -maxdepth 1` 2> /dev/null || true
 
-image-build: ## Build new docker image
-	docker build \
-		-t ghcr.io/stroeer/protoc-dockerized:latest \
-		-t ghcr.io/stroeer/protoc-dockerized:$(PROTOC_VERSION) \
+.PHONY: help
+help: ## Display this help screen
+	@grep -E '^[0-9a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | sort | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
+
+#######################
+# protoc docker image #
+#######################
+
+ACTOR = $(GITHUB_ACTOR)
+TOKEN = $(GITHUB_TOKEN)
+.PHONY: ghcr-login
+ghcr-login:
+	@echo "+ $@"
+	@echo $(TOKEN) | docker login ghcr.io -u $(ACTOR) --password-stdin
+
+.PHONY: protoc-build
+protoc-build: ## Build protoc docker image
+	@echo "+ $@"
+	@docker build \
+		--tag ghcr.io/stroeer/protoc-dockerized:latest \
+		--tag ghcr.io/stroeer/protoc-dockerized:$(PROTOC_VERSION) \
 		--build-arg PROTOC_VERSION=$(PROTOC_VERSION) .
 
-image-release: image-build ## Release new docker image to GHCR
-	docker push ghcr.io/stroeer/protoc-dockerized:latest
-	docker push ghcr.io/stroeer/protoc-dockerized:$(PROTOC_VERSION)
+.PHONY: protoc-push
+protoc-push: ghcr-login protoc-build ## Push protoc docker image to https://github.com/orgs/stroeer/packages/container/package/protoc-dockerized
+	@echo "+ $@"
+	@docker push --all-tags ghcr.io/stroeer/protoc-dockerized
+
+###########
+# release #
+###########
+
+DESCRIBE			= $(eval DESCRIBE := $(shell git fetch --all > /dev/null && git describe --match "v*" --always --tags))$(DESCRIBE)
```

Comment:
this is a quite expensive operation, that is only required when releasing. making this "lazy" increased the build time for `make generate` from `15 seconds` to `3 seconds`. #perf-matters